From bd82012a1a215b3d88bcc248cbcec3745f9aa6db Mon Sep 17 00:00:00 2001
From: Alessandro Astone <ales.astone@gmail.com>
Date: Sat, 4 May 2019 20:06:48 +0200
Subject: [PATCH 3/3] recovery: always wipe bootloader message from index 0

 * We use custom offsets to ignore unwanted flags from the bootloader,
   but we were also ignoring them when clearing the misc block,
   leaving it dirty after exiting recovery.
   This change allows to properly clear misc on devices that set
   a custom bootloader message offset.

Change-Id: I2d4e0702a2d8cbbef6274a87ce9499b0f69310dd
---
 bootloader_message/bootloader_message.cpp | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/bootloader_message/bootloader_message.cpp b/bootloader_message/bootloader_message.cpp
index 7ebd86a5..fa6b20e8 100644
--- a/bootloader_message/bootloader_message.cpp
+++ b/bootloader_message/bootloader_message.cpp
@@ -164,7 +164,11 @@ bool write_bootloader_message(const bootloader_message& boot, std::string* err)
 
 bool clear_bootloader_message(std::string* err) {
   bootloader_message boot = {};
-  return write_bootloader_message(boot, err);
+  std::string misc_blk_device = get_misc_blk_device(err);
+  if (misc_blk_device.empty()) {
+    return false;
+  }
+  return write_misc_partition(&boot, sizeof(boot), misc_blk_device, 0 /* offset */, err);
 }
 
 bool write_bootloader_message(const std::vector<std::string>& options, std::string* err) {
-- 
2.17.1

